juriscraper/pacer/appellate_attachment_page.py

```diff
@@ -0,0 +1,224 @@
+from __future__ import print_function
+
+import pprint
+import re
+import sys
+from typing import Dict
+
+from lxml import html
+
+from juriscraper.lib.html_utils import strip_bad_html_tags_insecure
+from juriscraper.lib.string_utils import force_unicode
+from juriscraper.pacer.utils import get_pacer_doc_id_from_doc1_url
+from .reports import BaseReport
+from ..lib.log_tools import make_default_logger
+
+logger = make_default_logger()
+
+
+class AppellateAttachmentPage(BaseReport):
+    """An object for querying and parsing the appellate att. page report.
+
+    * Some notes on Appellate attachement pages.
+    The document number for the main document is not identified as such in the
+    HTML.  Trial and error suggests it is the lowest numbered document
+    number.  This is not always the first document listed in the order of
+    attachments on the appellate attachment page.
+    """
+
+    PATH = "docs1/"
+
+    def __init__(self, court_id, pacer_session=None):
+        super(AppellateAttachmentPage, self).__init__(court_id, pacer_session)
+
+    def query(self, document_number):
+        """Query the "attachment page" endpoint and set the results to self.response.
+
+        The appellate attachment page uses the a variable called dls_id
+        in the URL path which appears to be a variation of the main document
+        id number.  We can generate the DLS ID by changing the fourth number
+        in the document ID to 0.
+        :param document_number: The internal PACER document ID for the item.
+        :return: a request response object
+        """
+        assert (
+            self.session is not None
+        ), "session attribute of DocketReport cannot be None."
+
+        # Generate the document URL from the document number.
+        document_number = document_number[:3] + "0" + document_number[4:]
+        url = self.url + document_number
+        logger.info(u"Querying the attachment page endpoint at URL: %s", url)
+        self.response = self.session.get(url)
+        self.parse()
+
+    def _strip_bad_html_tags_insecure(self, tree):
+        """Remove bad tags from HTML while leaving scripts.
+
+        :param tree: A tree you wish to cleanup
+        :type tree: lxml.html.HtmlElement
+
+        :return: None, instead sets self.tree to a cleaned lxml.html.HtmlElement.
+        """
```

Comment:
I think I'd suggest something simpler here, even though I know I'm constantly bugging you for more detailed docstrings. Perhaps: 

```suggestion
        """Override base function to include scripts."""
```

Reply:
This seems solved.